### PR TITLE
[PDI-19020] - White Spaces are ignored in a JSON Input step

### PR DESCRIPTION
**Pull request by bcostahitachivantara**
@pentaho-lmartins 
@ssamora 
@smmribeiro

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 24m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **5005**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7796/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/di/core/Const.java [hunk 1]
```diff
@@ -754,6 +754,12 @@ public String getMessage() {
    */
   public static final String KETTLE_DO_NOT_NORMALIZE_NULL_STRING_TO_EMPTY = "KETTLE_DO_NOT_NORMALIZE_NULL_STRING_TO_EMPTY";
 
+
+  /**
+   * This flag will force to return the original string with only spaces instead of an empty string.
+   */
+  public static final String KETTLE_FORCE_STRING_WITH_SPACES_ONLY_AS_NOT_EMPTY = "KETTLE_FORCE_STRING_WITH_SPACES_ONLY_AS_NOT_EMPTY";
```
Comment (ssamora) [inline core/src/main/java/org/pentaho/di/core/Const.java:761]: i would call it something like the kettle property above. 
Something like:
KETTLE_DO_NOT_NORMALIZE_SPACES_ONLY_STRING_TO_EMPTY

What do you think?

**Review by smmribeiro — CHANGES_REQUESTED** [review]
Don't forget to add the description of the newly created Kettle property to /engine/src/main/resources/kettle-variables.xml

File: core/src/main/java/org/pentaho/di/core/Const.java (hunk 1 above)
Comment (bcostahitachivantara) [reply core/src/main/java/org/pentaho/di/core/Const.java:761]: :+1:

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 33m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,engine
```
---
:ok_hand: All tests passed!
 
Tests run: **5005**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7796/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 24s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,engine
```
---
:ok_hand: All tests passed!
 
Tests run: **5005**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7796/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**